modules/dop/providers/guide/guide.service.go

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package guide
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"github.com/erda-project/erda-proto-go/dop/guide/pb"
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle"
+	"github.com/erda-project/erda/modules/dop/providers/guide/db"
+	"github.com/erda-project/erda/modules/dop/services/apierrors"
+	"github.com/erda-project/erda/pkg/common/apis"
+	"github.com/erda-project/erda/pkg/limit_sync_group"
+)
+
+const (
+	DicePipelinePath string = ".dice/pipelines"
+	ErdaPipelinePath string = ".erda/pipelines"
+	InitCommitID     string = "0000000000000000000000000000000000000000"
+	BranchPrefix     string = "refs/heads/"
+)
+
+type GuideService struct {
+	bdl *bundle.Bundle
+	db  *db.GuideDB
+}
+
+func (g *GuideService) CreateGuideByGittarPushHook(ctx context.Context, req *pb.GittarPushPayloadEvent) (*pb.CreateGuideResponse, error) {
+	if req.Content.Before != InitCommitID {
+		return &pb.CreateGuideResponse{}, nil
+	}
+
+	orgID, err := strconv.ParseUint(req.OrgID, 10, 64)
+	if err != nil {
+		return nil, apierrors.ErrCreateGuide.InvalidParameter(err)
+	}
+	projectID, err := strconv.ParseUint(req.ProjectID, 10, 64)
+	if err != nil {
+		return nil, apierrors.ErrCreateGuide.InvalidParameter(err)
+	}
+	appID, err := strconv.ParseUint(req.ApplicationID, 10, 64)
+	if err != nil {
+		return nil, apierrors.ErrCreateGuide.InvalidParameter(err)
+	}
+	guide := db.Guide{
+		// TODO add jump link
+		JumpLink:      "",
```

Comment:
this jumpLink should be completed before merged

Reply:
it is no need `JumpLink`